packages/docs/modules/ROOT/pages/network-js.adoc

```diff
@@ -0,0 +1,302 @@
+
+import Web3Info from './components/Web3Info/index.js';
+
+// const infuraProjectId = process.env.REACT_APP_INFURA_PROJECT_ID;
+const infuraProjectId = '95202223388e49f48b423ea50a70e336';
+
+function App() {
+
+  const web3Context = useWeb3(`wss://mainnet.infura.io/ws/v3/${infuraProjectId}`);
+
+  return (
+    <div className="App">
+      <div>
+        <h1>OpenZeppelin Network.js</h1>
+        <Web3Info title="Web3 Info" web3Context={web3Context} />
+      </div>
+    </div>
+  );
+}
+
+export default App;
+----
+
+Start our dapp again by running `npm start` from within the `client` directory. 
+
+*OpenZeppelin Network.js* re-renders React components on changes such as account, network, and connection.
+
+Try changing networks in MetaMask and *OpenZeppelin Network.js* will cause the component to update with the network ID and network name.  
+
+== Add request access to injected web3 to the dapp
+For the third iteration of our dapp we will add a mechanism request to access injected web3 and once the user has connected will display the address of the users account.
+
+NOTE: Best practice is to wait to request injected web3 access until a user wants to perform an action that they can only do with injected web3.  See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1102.md[EIP-1102] for details.
+Remember dapps can be configured to use multiple web3 providers such as a network provider if necessary.
+
+In the code below, we get `accounts` from the web3Context and if available the users account is displayed in the dapp.
+If `accounts` are not available we display a button for the user to request access to injected web3.  When pressed, the `requestAuth` function in web3Context is called and the injected web3 provider can display a dialog to the user requesting access.
+We also use react functionality, `useCallback` to setup the callback for request access.
+
+On the Web3Info components `/Web3Info/index.js` file, replace `index.js` with the following code:
+
+[source,js]
+----
+import React, { useCallback } from 'react';
+
+export default function Web3Info(props) {
+  const { web3Context } = props;
+  const { networkId, networkName, accounts, providerName } = web3Context;
+
+  const requestAuth = async web3Context => {
+    try {
+      await web3Context.requestAuth();
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const buttonStyle = {
+    color: 'blue',
+    cursor: 'pointer',
+  };
+
+  const requestAccess = useCallback(() => requestAuth(web3Context), []);
+
+  return (
+    <div>
+      <h3> {props.title} </h3>
+      <div>Network: {networkId ? `${networkId} – ${networkName}` : 'No connection'}</div>
+      <div>Your address: {accounts && accounts.length ? accounts[0] : 'Unknown'}</div>
+      <div>Provider: {providerName}</div>
+      {accounts && accounts.length ? (
+        <div>Accounts & Signing Status: Access Granted</div>
+      ) : !!networkId && providerName !== 'infura' ? (
+        <div>
```

Comment:
I change this to a `<button>` element and remove the `buttonStyle` variable. I feel that it distracts from the main point.

Reply:
I think the starter kit uses a Button component and I couldn't work out how to create a button easily so did button style (rather than using a `button` element).  Oops.  I really need to learn React.

It looks better now!  Thank you.